src/index/bwtree_index.cpp

```diff
@@ -78,23 +73,22 @@ BWTREE_INDEX_TYPE::InsertEntry(const storage::Tuple *key,
  * If the key-value pair does not exists yet in the map return false
  */
 BWTREE_TEMPLATE_ARGUMENTS
-bool
-BWTREE_INDEX_TYPE::DeleteEntry(const storage::Tuple *key,
-                               const ItemPointer &location) {
+bool BWTREE_INDEX_TYPE::DeleteEntry(const storage::Tuple *key,
+                                    const ItemPointer &location) {
   KeyType index_key;
   index_key.SetFromKey(key);
-  
+
   // Must allocate new memory here
   ItemPointer *ip_p = new ItemPointer{location};
-  
+
   // In Delete() since we just use the value for comparison (i.e. read-only)
   // it is unnecessary for us to allocate memory
   bool ret = container.DeleteExchange(index_key, &ip_p);
-  
+
   // IF delete succeeds then DeleteExchange() will exchange the deleted
   // value into this variable
-  if(ret == true) {
-    //delete ip_p;
+  if (ret == true) {
+    // delete ip_p;
```

Comment:
It is commented out to avoid malloc-free mismatching, since now the interface is a little bit inconsistent, in a sense that the index is declared as template type <GenericKey, ItemPointer *>, but the actual value given to the index wrapper InsertEntry() and DeleteEntry() is class ItemPointer. This is generally not accepted use pattern for an index whose ValueType is ItemPointer *.

I will file an issue and resolve this gradually after we have other things stabilized.